src/bootstrap/src/core/builder.rs

```diff
@@ -1273,7 +1274,7 @@ impl<'a> Builder<'a> {
             self.clear_if_dirty(&out_dir, &backend);
         }
 
-        if cmd == "doc" || cmd == "rustdoc" {
+        if cmd == "doc" || cmd == "rustdoc" && !CiEnv::is_ci() {
```

Comment:
I think this shouldn't cause any problem? I assume that the runners are automatically purging the entire build related state upon the termination.